### DISCUSSION TITLE
vite-plugin-eslint implementation with nuxt3

### DISCUSSION THREAD

**Discussion by Krystus** [nuxt/nuxt#16694] [Questions]
Is there a way to implement eslint errors with Vite and nuxt3?

Something like that seemes to take no effect.

```
import eslintPlugin from "vite-plugin-eslint"

export default defineNuxtConfig({
  vite: {
    plugins: [
      eslintPlugin(),
    ],
  },
```

**Comment by danielroe** [comment]
I believe @ricardogobbosouza is working to migrate https://github.com/nuxt-community/eslint-module to support Nuxt 3.

**Comment by ricardogobbosouza** [comment]
Hi @Krystus
I'm working on [eslint-module](https://github.com/nuxt-community/eslint-module) support for nuxt3 which uses vite/webpack.

In your case you need to configure the include option
https://github.com/gxmari007/vite-plugin-eslint#include
Note: It is worth remembering that errors/warnings are only printed in the terminal

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt3'
import eslintVitePlugin from "vite-plugin-eslint"

export default defineNuxtConfig({
  vite: {
    plugins: [
      eslintVitePlugin({
        include: ['./**/*.vue']
      }),
    ],
  }
})

**Comment by ricardogobbosouza** [comment]
Starting support nuxt3/bridge
https://github.com/nuxt-community/eslint-module/pull/70

**Comment by bf** [comment]
@Krystus have you found a good solution for this problem?

**Comment by dpmango** [comment]
Working for me

```js
import eslintPlugin from 'vite-plugin-eslint'

export default defineNuxtConfig({
  vite: {
    plugins: [
      eslintPlugin({
        failOnError: false,
        cache: false,
      }),
    ],
  }
}
```

`eslintrc.js`
```js
module.exports = {
  root: true,
  env: {
    browser: true,
    es2021: true,
    node: true,
  },
  ignorePatterns: ['.nuxt/', '.output/', 'dist/', 'public/', '**/*.min.js'],
  extends: [
    '@nuxtjs/eslint-config-typescript',
    'eslint:recommended',
    'plugin:vue/vue3-recommended',
    'plugin:nuxt/recommended',
  ],
  rules: {
  },
}
```



Anything to do with increased build times?

build times:
Done in 33.78s. - with vite-plugin-eslint
Done in 19.36s. - without

One page just for now, but eslint cli is almost 2x times faster
```
$ eslint . -c .eslintrc.js --ext ".js,.ts,.vue"
✨  Done in 7.89s.
```
